include/seqan3/core/algorithm/configuration.hpp

```diff
@@ -101,19 +101,17 @@ class configuration : public std::tuple<configs_t...>
     ~configuration()                                           = default; //!< Defaulted.
 
     /*!\brief Constructs a configuration from a single configuration element.
-     * \param elem The element to store.
+     * \tparam config_element_t The configuration element to add; must model
+     *                          seqan3::detail::config_element_specialisation.
+     * \param[in] config_element The configuration element to construct the configuration from.
      */
-    template <typename derived_t, typename value_t>
-    constexpr configuration(pipeable_config_element<derived_t, value_t> && elem) :
-        base_type{static_cast<derived_t &&>(std::move(elem))}
-    {}
-
-    /*!\brief Constructs a configuration from a single configuration element.
-     * \param elem The element to store.
-     */
-    template <typename derived_t, typename value_t>
-    constexpr configuration(pipeable_config_element<derived_t, value_t> const & elem) :
-        base_type{static_cast<derived_t const &>(elem)}
+    template <typename config_element_t>
+    //!\cond
+        requires (!std::same_as<std::remove_cvref_t<config_element_t>, configuration>) &&
+                 detail::config_element_specialisation<std::remove_cvref_t<config_element_t>>
```

Comment:
```suggestion
        requires (!std::same_as<config_element_t, configuration>) &&
                 detail::config_element_specialisation<config_element_t>
```

Reply:
Thank you for the review 👍 
The remove cvref is actually necessary because the function parameter is captured as a forwarding reference. 
This means that it can eiter capture a temporary as an rvalue reference `obj &&` or an lvalue as regular reference `obj &`. Now the real fun starts with the reference collapsing rules. If `config_element_t` (or I use it here short `t`), is an lvalue reference we have following substitution `t &` `&& obj` which collapses to `t &` `obj` and the reference is part of the template parameter. If it is an rvalue reference we have `t &&` `&& obj` which collapses to `t` `&& obj` so the template parameter does not contain a reference anymore. Accordingly, the interface needs to remove the reference to compare it with the concrete type `configuration` which neither contains const nor references. Also the config_element_specialisation concept does not strip away the reference, since that has semantically different menaings. The most obvious one is that a reference is not an object, such that an object has member functions but a reference does not. So one requirement of that concept is that the type is semiregular which is not true for references. You could argue, that the concept could remove this, but then the caller cannot differentiate whether he has a pure object or a reference thereof. So if the caller would create an default constructed instance of that type, it would not work anymore, since a reference cannot be default constructed. So the concept would give the wrong impression.

Reply:
Yes! you are right.
I have mixed up class template with the template of the constructor. For what ever reason I thought the type deduction determines the constructor template, but it only decides the class template.